src/task2/flattening_module/imageflattener.py

```diff
@@ -109,47 +112,85 @@ def _flatten_img(self):
 
         flat = nPTransform.four_points_correct_aspect(self.movable_image.cv_img,
                                                       rect,
-                                                      self.dim[0],
-                                                      self.dim[1])
+                                                      self.__selected_panel.dims[0],
+                                                      self.__selected_panel.dims[1])
 
         self.movable_image.cv_img = flat
         self.movable_image.reset()
         self.movable_image.clear_dots()
 
         self.__clicks = []
 
+    def _get_panel_within_clicks(self) -> Panel:
+        """returns the most central panel within the bounds of the clicked area of a panel
+        with all attributes set to None if no such panel exists.
+
+        Precondition:
+        len(self.__clicks) == 4
+        """
+        if any(p.pixel is None for p in self.__panels_in[self.curr_path]):
+            return Panel(None, None)
+
+        bounds = Polygon([self.movable_image.canvas_to_cv(click) for click
+                          in self.__clicks])
+
+        panels_in_bounds = []
+
+        for panel in self.__panels_in[self.curr_path]:
+            p = Point(panel.pixel)
+            print(panel.pixel)
+            print(self.__clicks)
+            if p.within(bounds):
+                panels_in_bounds.append(panel)
+
+        if panels_in_bounds == []:
+            return Panel(None, None)
+        else:
```

Comment:
Not a requested change but a coding style I find nice: 

Nested shit is nasty. In this case, we already exit early with the `if` statement. We can just forgo the else and move all that code up one level 😄